pkgs/servers/icepeak/default.nix

```diff
@@ -0,0 +1,22 @@
+{ callPackage, fetchFromGitHub, stdenvNoCC }:
+let
+    src = fetchFromGitHub {
+      owner  = "channable";
+      repo   = "icepeak";
+      rev    = "v0.7.0";
+      sha256 = "1f9y4cjkwwc5r7wjmbnwnif7qfc3nxkgvrg361scjcsb55d777sl";
+    };
+
+    # icepeak itself is built with Nix, so we can use callPackage here
+    icepeak = callPackage "${src}/server" {};
```

Comment:
this doesn't work
```
cannot import '/nix/store/p4gzy8wbc92cmj0q1r10ksaws9w9nh4f-source/server', since path '/nix/store/mw17npnlqsmfbwbcm512km0wck900b5w-source.drv' is not valid, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/hetzner-sb65-1083082-hel1-dc2/lib/customisation.nix:119:45
```
Also, having a default.nix in a repo is fine, but the responsibility for knowing how to build a package in nixpkgs should be nixpkgs, not deferred to the package. (personal opinion)

@Infinisil  opionion?

Reply:
I understand the point of nixpkgs being the source of truth here. It is too bad that this doesn't work though, since that means that I'll have to copy the contents from the `default.nix` in the Icepeak repo to this file here (and repeat that process whenever we change the `default.nix` in the Icepeak repo).

I assumed that there would be some way to do this, since e.g. for packages that contain a Makefile you essentially also delegate the actual buildsteps to that external file.

Reply:
correct, but the main difference is that make is dumb, it will just make assumptions about the commands that can be executed, and as part of our packaging, we just fulfill those assumptions.

delegating the construction of the nix expression to the package creates the scenario where the package is making assumptions about what's contained in nixpkgs, and nixpkgs is making some assumptions about what it needs to provide for the package. Which blurs the lines of responsibility.

Furthermore, nixpkgs has many treewide changes, and if we wanted to do something like fix some flags or metadata, then we would need to open a PR against your repo, then update the src in nixpkgs, which doesn't seem right.

Reply:
Ah okay, that makes sense. I'll update the PR to not depend on an external Nix file. Thanks for the explanation!